lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,223 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  static get INCLUDE_INCOME_TRANSACTIONS() {
+    return 1 >> 1
+  }
+
+  static get INCLUDE_OUTCOME_TRANSACTIONS() {
+    return 1 >> 2
+  }
+
+  static get INCLUDE_ALL_TRANSACTIONS() {
+    CardanoAPI.INCLUDE_INCOME_TRANSACTIONS |
+      CardanoAPI.INCLUDE_OUTCOME_TRANSACTIONS
+  }
+
+  static get PER_PAGE_MAXIMUM() {
+    return 50
+  }
+
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, returnFullResponse = false) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return this.client({
+      method,
+      url,
+      params,
+      data
+    }).then(response => {
+      if (response.error) throw response.error
+
+      if (returnFullResponse) {
+        return response
+      }
+
+      return response.data.data
+    })
+  }
+
+  async fetchAllPages(endpoint, params, data) {
+    const fetchAllPagesAccumular = (page, dataAccumular = []) =>
+      this.fetch(endpoint, { ...params, page }, data, true).then(response => {
+        const responseData = Array.isArray(response.data.data)
```

Comment:
Are we sure there exists endpoints which
1) use paging
2) at the same time, won't return an array?

Reply:
Yes, pagination is in all responses - f.e. https://cardanodocs.com/technical/wallet/api/v1/?v=1.6.0#tag/Info - i don't suppose anybody will run `fetchAllPages` on endpoints with non-array data, but i take this as edge case

Reply:
Then just assert. I would rather have defensive code than one multi-purpose blob

Reply:
ok